src/unibasicTemplates/subroutines/external/increment.njk

```diff
@@ -0,0 +1,189 @@
+  go returnFromSub
+end
+
+* get the number of retries to perform if the record is locked
+if udoGetProperty(options, 'retry', retryMax, type) then
+  call error_handler(ERROR_MALFORMED_INPUT, output)
+  go returnFromSub
+end
+
+* get the retry delay between retries
+if udoGetProperty(options, 'retryDelay', retryDelay, type) then
+  call error_handler(ERROR_MALFORMED_INPUT, output)
+  go returnFromSub
+end
+
+* get the array of operations from the options
+if udoGetProperty(options, 'operations', operations, type) then
+  call error_handler(ERROR_MALFORMED_INPUT, output)
+  go returnFromSub
+end
+
+operationsCount = 0
+begin case
+  case type ne UDO_ARRAY
+    call error_handler(ERROR_MALFORMED_INPUT, output)
+    go returnFromSub
+  case udoArrayGetSize(operations, operationsCount) ne UDO_SUCCESS
+    call error_handler(ERROR_UDO, output)
+    go returnFromSub
+end case
+
+* make sure the array of operations passed in wasn't empty
+if operationsCount eq 0 then
+  call error_handler(ERROR_MALFORMED_INPUT, output)
+  go returnFromSub
+end
+
+* open the file
+open filename to f.file on error
+  call error_handler(ERROR_FILE_OPEN, output)
+  go returnFromSub
+end else
+  call error_handler(ERROR_FILE_OPEN, output)
+  go returnFromSub
+end
+
+* read and lock the record from the file
+tries = 0
+
+loop
+  readu record from f.file, recordId on error
+    call error_handler(ERROR_RECORD_READ, output)
+    go closeAndReturnFromSub
+  end locked
+    if tries > retryMax then
+      call error_handler(ERROR_RECORD_LOCKED, output)
+      go closeAndReturnFromSub
+    end else
+      tries+= 1
+      sleep retryDelay
+    end
+  end then
+    * lock successful get out of read lock loop
+    exit
+  end else
+    * record not found
+    call error_handler(ERROR_RECORD_NOT_FOUND, output)
+    go closeReleaseAndReturnFromSub
+  end
+repeat
+
+* increment fields
+processCount = 0
+loop while processCount lt operationsCount
+  processCount += 1
+
+  * get the next operation from the array
+  if udoArrayGetNextItem(operations, operation, type) ne UDO_SUCCESS then
+    call error_handler(ERROR_UDO, output)
+    go closeReleaseAndReturnFromSub
+  end
+
+  * get the path
+  if udoGetProperty(operation, 'path', path, type) then
+    call error_handler(ERROR_MALFORMED_INPUT, output)
+    go closeReleaseAndReturnFromSub
+  end
+  * path must be a string
+  if type ne UDO_STRING then
+    call error_handler(ERROR_MALFORMED_INPUT, output)
+    go closeReleaseAndReturnFromSub
+  end
+
+  * get the value to increment by
+  if udoGetProperty(operation, 'value', value, type) then
+    call error_handler(ERROR_MALFORMED_INPUT, output)
+    go closeReleaseAndReturnFromSub
+  end
+  * value must be numeric
+  if type ne UDO_NUMBER then
+    call error_handler(ERROR_MALFORMED_INPUT, output)
+    go closeReleaseAndReturnFromSub
+  end
+
+  * parse out path.
+  pathMatchTest = path match "1n'.'1n'.'1n"
+  if pathMatchTest eq 0 then
```

Comment:
This is returning a 1 or a 0 which is a boolean true/false in basic. It should be treated as boolean for clarity.

You can consolidate this into a single line expression:
`if not(path matches "1n'.'1n'.'1n") then`

Reply:
Done thank you.